### DISCUSSION TITLE
WS Benchmarks

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#185] [Ideas]
Can the performance be benchmarked and tracked against frameworks like:
- https://github.com/obsidiandynamics/socketx
- https://babl.ws/
and improved if needed.

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath 
Our WebSocket implementation is a thin abstraction on top of Java’s `SocketChannel`. It would not make much sense to compare the performance to other frameworks as most of them process WebSocket in a similar way. A bottleneck, in this case, would be the actual sending of WebSocket frames via a network, rather than encoding/decoding frames which is quite straightforward.
